libCacheSim/bin/cachesim/cache_init.h

```diff
@@ -12,60 +12,90 @@
+  static const eviction_algo_entry_t simple_algos[] = {
+      {"lru", LRU_init},
+      {"fifo", FIFO_init},
+      {"arc", ARC_init},
+      {"arcv0", ARCv0_init},
+      {"lhd", LHD_init},
+      {"random", Random_init},
+      {"randomTwo", RandomTwo_init},
+      {"lfu", LFU_init},
+      {"gdsf", GDSF_init},
+      {"lfuda", LFUDA_init},
+      {"twoq", TwoQ_init},
+      {"2q", TwoQ_init},
+      {"slru", SLRU_init},
+      {"slruv0", SLRUv0_init},
+      {"lecar", LeCaR_init},
+      {"lecarv0", LeCaRv0_init},
+      {"RandomLRU", RandomLRU_init},
+      {"cacheus", Cacheus_init},
+      {"size", Size_init},
+      {"lfucpp", LFUCpp_init},
+      {"wtinyLFU", WTinyLFU_init},
+      {"nop", nop_init},
+      {"fifo-reinsertion", Clock_init},
+      {"clock", Clock_init},
+      {"second-chance", Clock_init},
+      {"clockpro", ClockPro_init},
+      {"lirs", LIRS_init},
+      {"fifomerge", FIFO_Merge_init},
+      {"fifo-merge", FIFO_Merge_init},
+      {"flashProb", flashProb_init},
+      {"sfifo", SFIFO_init},
+      {"sfifov0", SFIFOv0_init},
+      {"lru-prob", LRU_Prob_init},
+      {"fifo-belady", FIFO_Belady_init},
+      {"lru-belady", LRU_Belady_init},
+      {"sieve-belady", Sieve_Belady_init},
+      {"s3lru", S3LRU_init},
+      {"s3fifo", S3FIFO_init},
+      {"s3-fifo", S3FIFO_init},
+      {"s3fifov0", S3FIFOv0_init},
+      {"s3-fifov0", S3FIFOv0_init},
+      {"s3fifod", S3FIFOd_init},
+      {"qdlp", QDLP_init},
+      {"CAR", CAR_init},
+      {"sieve", Sieve_init},
+  };
+
+  cache_t *(*init_func)(common_cache_params_t, const char *) = NULL;
+  for (size_t i = 0; i < sizeof(simple_algos) / sizeof(simple_algos[0]); ++i) {
+    if (strcasecmp(eviction_algo, simple_algos[i].name) == 0) {
+      init_func = simple_algos[i].init_func;
+      break;
+    }
+  }
+
+  // Initializing for algorithms which require special handling (not in
+  // simple_algos)
+  if (init_func) {
+    cache = init_func(cc_params, eviction_params);
   } else if (strcasecmp(eviction_algo, "hyperbolic") == 0) {
     cc_params.hashpower = MAX(cc_params.hashpower - 8, 16);
     cache = Hyperbolic_init(cc_params, eviction_params);
-  } else if (strcasecmp(eviction_algo, "lecar") == 0) {
-    cache = LeCaR_init(cc_params, eviction_params);
-  } else if (strcasecmp(eviction_algo, "lecarv0") == 0) {
-    cache = LeCaRv0_init(cc_params, eviction_params);
-  } else if (strcasecmp(eviction_algo, "RandomLRU") == 0) {
-    cache = RandomLRU_init(cc_params, eviction_params);
-  } else if (strcasecmp(eviction_algo, "cacheus") == 0) {
-    cache = Cacheus_init(cc_params, eviction_params);
-  } else if (strcasecmp(eviction_algo, "size") == 0) {
-    cache = Size_init(cc_params, eviction_params);
-  } else if (strcasecmp(eviction_algo, "lfucpp") == 0) {
-    cache = LFUCpp_init(cc_params, eviction_params);
   } else if (strcasecmp(eviction_algo, "tinyLFU") == 0) {
     if (eviction_params == NULL) {
       cache = WTinyLFU_init(cc_params, eviction_params);
```

Comment:
Consider one corner case, user pass in eviction_params as an empty string `""`. Then the parser will crash with ",window-size=0.01". Can we become robust for this corner case?